homeassistant/components/rest/binary_sensor.py

```diff
@@ -101,16 +104,17 @@ def __init__(
             unique_id=unique_id,
         )
         self._previous_data = None
-        self._value_template = config.get(CONF_VALUE_TEMPLATE)
+        self._value_template: Template | None = config.get(CONF_VALUE_TEMPLATE)
         if (value_template := self._value_template) is not None:
             value_template.hass = hass
 
         self._attr_device_class = config.get(CONF_DEVICE_CLASS)
 
-    def _update_from_rest_data(self):
+    def _update_from_rest_data(self) -> None:
         """Update state from the rest data."""
         if self.rest.data is None:
             self._attr_is_on = False
+            return
```

Comment:
Slight logic change here, but I think that was the original intention.
Without this, then response is `str | None` and `mypy` complains.